### ISSUE TITLE
Improve custom protocol detector to work better on safari & edge

### ISSUE THREAD

**Issue by larrysalibra** [stx-labs/stacks.js#298]
https://github.com/larrysalibra/custom-protocol-detection

**Comment by larrysalibra** [comment]
I've done a bit of research into detecting custom protocols on Safari both on iOS and macOS.

The TL;DR is that Apple seems to have done a bunch of work to make it not possible to detect protocol handlers in Safari. 

In safari, on both iOS and macOS, the best it seems we can do is to redirect users to a web page with instructions to follow if the app doesn't open, and then try to open the `blockstack:` link in an iframe. This usually (but not always on in my testing on iOS) suppresses the unhelpful, unsupported error dialog/page.

The way to detect custom handlers in iOS apps is to use this method: https://developer.apple.com/documentation/uikit/uiapplication/1622952-canopenurl

In iOS safari, the preferred way to handle something by an app would be using universal links instead of a custom protocol handler. https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html
